rust/kcl-lib/src/std/clone.rs

```diff
@@ -234,13 +236,30 @@ async fn fix_sketch_tags_and_references(
         }
     }
 
+    // Map the surface tags to the new surface ids.
+    let mut surface_id_map: HashMap<String, &ExtrudeSurface> = HashMap::new();
+    let surfaces = surfaces.unwrap_or_default();
+    for surface in surfaces.iter() {
+        if let Some(tag) = surface.get_tag() {
+            surface_id_map.insert(tag.name.clone(), surface);
+        }
+    }
+
     // Fix the tags
     // This is annoying, in order to fix the tags we need to iterate over the paths again, but not
     // mutable borrow the paths.
     for path in new_sketch.paths.clone() {
         // Check if this path has a tag.
         if let Some(tag) = path.get_tag() {
-            new_sketch.add_tag(&tag, &path, exec_state);
+            let mut surface = None;
+            if let Some(found_surface) = surface_id_map.get(&tag.name) {
+                let mut new_surface = (*found_surface).clone();
+                let new_face_id = entity_id_map.get(&new_surface.face_id()).copied().unwrap_or_default();
```

Comment:
This `unwrap_or_default()` still seems weird to me because the default value of a `Uuid` is the UUID made of all zeros.